misc/factorial_recursive.c

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file factorial_recursive.c
+ * @brief factorial with recursive function call
```

Comment:
Provide a Wikipedia link in Markdown format (if available/possible).
If there's no Wikipedia link, add another web source for algorithm explanation. 🙂